adventurelib.py

```diff
@@ -115,6 +115,35 @@ def _match_context(context, active_context):
     )
 
 
+class TabCompleter(object):
+    """Class, used if `readline` is available for Tab/Auto Comlpetion"""
+    def __init__(self, commands):
+        self.commands = commands
+
+    def complete(self, text, state):
+        space = re.compile('.*\\s+$', re.M)
+        "Generic readline completion entry point."
+        buffer = readline.get_line_buffer()
+        line = readline.get_line_buffer().split()
+        # show all commands
+        if not line:
+            return [c + ' ' for c in self.commands][state]
+        # account for last argument ending in a space
+        if space.match(buffer):
+            line.append('')
+        # resolve command to the implementation function
+        cmd = line[0].strip()
+        if cmd in self.commands:
+            impl = getattr(self, 'complete_%s' % cmd)
+            args = line[1:]
+            if args:
+                return (impl(args) + [None])[state]
+            return [cmd + ' '][state]
+        results = [c + ' ' for c in self.commands if c.startswith(cmd)] \
+            + [None]
+        return results[state]
```

Comment:
All thanks to [this](https://stackoverflow.com/questions/5637124/tab-completion-in-pythons-raw-input) thread.